content/articles/2019-09-tidyr-1-0-0.Rmarkdown

```diff
@@ -0,0 +1,316 @@
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+options(
+  tibble.print_min = 6, 
+  tibble.print_max = 6
+)
+```
+
+I'm very excited to announce the release of [tidyr 1.0.0](https://tidyr.tidyverse.org)! tidyr provides a set of tools for transforming data frames to and from [tidy data](https://tidyr.tidyverse.org/dev/articles/tidy-data.html), where each variable is a column and each observation is a row. Tidy data is a convention for matching the semantics and structure of your data that makes using the rest of the tidyverse (and many other R packages) much easier. 
+
+Install tidyr with:
+
+```{r, eval = FALSE}
+install.packages("tidyr")
+```
+
+As you might guess from the version number, this is a major release, and the 1.0.0 moniker indicates that I'm finally happy with the overall interface of the package. This has been a long time coming: it's five years since the first tidyr release, nine years since the first reshape2 release, and fourteen years since the first reshape release! 
+
+This blog post summarises the four major changes to the package:
+
+* New `pivot_longer()` and `pivot_wider()` supercede `spread()` and `gather()`. 
+  The new functions are substantially more powerful, thanks to ideas from the 
+  data.table and cdata packages, and I'm confident that you'll find them easier 
+  to use and remember than their predecessors thanks to careful consideration 
+  of their interfaces.
+  
+* New functions for **rectangling**, converting deeply nested lists into tidy
+  data frames: `unnest_auto()`, `unnest_longer()`, `unnest_wider()`, and
+  `hoist()`.
+  
+* Interface changes to `nest()` and `unnest()` to match an emerging consensus
+  for interface design across the tidyverse. I've also discovered that nesting
+  can be broken down into two simpler steps, called `pack()` and `chop()`.
+  
+* New `expand_grid()`, a variant of `base::expand.grid()`. This is a useful 
+  function to know about, but also serves as a good reason to talk about the 
+  important role that [vctrs](http://vctrs.r-lib.org/) plays behind the scenes. 
+  You shouldn't ever _have_ to leave about vctrs, but it brings improvements to 
+  consistency and performance. 
+
+As well as implementing the new features, I've spent considerable time on the documentation, including four major new vignettes:
+
+* [`vignette("pivot")`](https://tidyr.tidyverse.org/dev/articles/pivot.html), 
+  [`vignette("rectangle")`](https://tidyr.tidyverse.org/dev/articles/rectangle.html), 
+  and [`vignette("nest")`](https://tidyr.tidyverse.org/dev/articles/nest.html) 
+  provide detailed documentation and case studies of pivotting, rectangling, 
+  and nesting respectively.
+
+* [`vignette("in-packages")`](https://tidyr.tidyverse.org/dev/articles/in-packages.html)
+  provides best practices for using tidyr inside
+  another package, and provides detailed advice on working with multiple 
+  versions of tidyr if an interface change has affected your package.
```

Comment:
This ends up being a very subtle mention. Perhaps this vignette could be plugged and linked again in, e.g., the `nest()`/`unnest()` section, since that is the type of call that most of the affected packages need to refactor.

Reply:
Repeated below, and explicitly mentioned `_legacy()` which I had forgotten